src/plugins/intel_gpu/src/graph/loop.cpp

```diff
@@ -877,14 +877,14 @@ void loop_inst::concatenated_memory_mapping::concat_mem(const int64_t curent_ite
                         concat_mem_shape.to_string(), ") != curent_iterations(",
                         curent_iterations, ") * sliced_mem_shape[axis](", sliced_mem_shape.to_string(), ")");
     std::vector<ov::Shape> shapes_to_concat(curr_iters, sliced_mem_shape);
-    std::vector<const char*> pointers_to_data(curr_iters);
+    std::vector<const void*> pointers_to_data(curr_iters);
     for (size_t i = 0; i < curr_iters; i++) {
         auto mem = sliced_mems[i];
-        pointers_to_data[stride > 0 ? i : (curr_iters - i - 1)] = reinterpret_cast<const char*>(mem->lock(stream));
+        pointers_to_data[stride > 0 ? i : (curr_iters - i - 1)] = reinterpret_cast<const void*>(mem->lock(stream));
     }
 
-    char* concat_data = reinterpret_cast<char*>(concatenated_mem->lock(stream));
-    ov::reference::concat(pointers_to_data, concat_data, shapes_to_concat, concat_mem_shape, axis, elem_size);
+    void* concat_data = reinterpret_cast<void*>(concatenated_mem->lock(stream));
```

Comment:
Pointer implicitly cast to `void*`.
```suggestion
    void* concat_data = concatenated_mem->lock(stream);
```

Reply:
This change has been reverted.